src/reducers/index.js

```diff
@@ -57,9 +57,27 @@ export const removeResource = (state, action) => {
   return validate(newState)
 }
 
+// shameless green -- nearly duplicates setRetrieveError()
+export const setPublishError = (state, action) => {
+  // Use safe navigation for dynamic properties: https://github.com/tc39/proposal-optional-chaining#syntax
+  const reason = action.payload?.reason
+  const newState = { ...state }
+
+  let serverError = 'There was a problem saving the resource'
+  if (reason) {
+    serverError += `: ${reason}`
+  } else {
+    serverError += '.'
+  }
+
+  newState.editor.serverError = serverError
+  return newState
+}
+
 export const setRetrieveError = (state, action) => {
-  const resourceTemplateId = action.payload.resourceTemplateId
-  const reason = action.payload.reason
+  // Use safe navigation for dynamic properties: https://github.com/tc39/proposal-optional-chaining#syntax
+  const resourceTemplateId = action.payload?.resourceTemplateId
```

Comment:
Again, you don't need this.  Remember that this expands to a lot of low level code.